### PR TITLE
WIP: xpath support

### PR DESCRIPTION
**Pull request by vosmith**
In #69 it was mentioned that xpath support would be welcomed.  I've started some work on this to see if this is what you had in mind.  Let me know what you think!

### CODE REVIEW

**Comment by asciimoo** [comment]
@vosmith this is super awesome, thanks. I have not heared about `github.com/antchfx/xmlquery` before, but looks promising.
As I see, it isn't compatible with html, however @antchfx has a dedicated package for querying html: github.com/antchfx/htmlquery .

**Comment by vosmith** [comment]
Yeah no problem, I think the two libraries have a very similar API.  I'll switch it over and make the necessary adjustments.

**Comment by asciimoo** [comment]
@vosmith great, thank you! Tests and benchmarks would be appreciated. =]

**Comment by vosmith** [comment]
I converted from the xmlquery library to the htmlquery library.  Let me know what you think!

**Comment by asciimoo** [comment]
@vosmith thanks for the changes. Something went wrong with the rebase and the PR contains previous commits, please fix it. Also, what about using `xmlquery` for XML content type and `htmlquery` for HTML content inside `handleOnXML`?

**Comment by vosmith** [comment]
Yeah i rebased to the master branch and it pulled in all of those commits, I'll try to get rid of them to clean this up.

Also I do like the idea of using different parsers based on the content-type.  I might be able to do some example crawling sitemaps as well.  Like Shopify https://www.shopify.com/sitemap.xml.  Let me see what I can do.

**Comment by vosmith** [comment]
I've updated the `handleOnXML` to support `xmlquery` for `text/xml` content types and `htmlquery` for `text/html` content types.  I've also added the Shopify sitemap example in the` _examples/sitemap` directory.  Let me know what you think!

File: colly.go [hunk 1]
```diff
@@ -688,6 +732,50 @@ func (c *Collector) handleOnHTML(resp *Response) {
 	}
 }
 
+func (c *Collector) handleOnXML(resp *Response) {
+	if (!strings.Contains(strings.ToLower(resp.Headers.Get("Content-Type")), "html") && !strings.Contains(strings.ToLower(resp.Headers.Get("Content-Type")), "xml")) || len(c.xmlCallbacks) == 0 {
+		return
+	}
+
+	if !strings.Contains(strings.ToLower(resp.Headers.Get("Content-Type")), "html") {
```
Comment (asciimoo) [inline colly.go]: This condition is a bit unclear for me. Why do you negate the expression? As I understand this block is executed if the content-type doesn't contain "html".

File: colly.go [hunk 2]
```diff
@@ -688,6 +732,50 @@ func (c *Collector) handleOnHTML(resp *Response) {
...
+				if c.debugger != nil {
+					c.debugger.Event(createEvent("xml", resp.Request.ID, c.ID, map[string]string{
+						"selector": cc.Query,
+						"url":      resp.Request.URL.String(),
+					}))
+				}
+				cc.Function(e)
+			})
+		}
+	} else if !strings.Contains(strings.ToLower(resp.Headers.Get("Content-Type")), "xml") {
```
Comment (asciimoo) [inline colly.go]: I have the same question here as in the previous block.

File: _examples/sitemap/sitemap.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package main
```
Comment (asciimoo) [inline _examples/sitemap/sitemap.go]: Perhaps this example could have a bit specific name, e.g. `shopify_sitemap`. `sitemap` is too general for me. What do you think?

File: colly.go (hunk 1 above)
Comment (vosmith) [reply colly.go]: Oh my! Good catch.

**Pull request merged**